src/core/server/http/http_server.test.ts

```diff
@@ -577,6 +578,173 @@ test('exposes route details of incoming request to a route handler', async () =>
+    {
+      path: '/',
+      validate: { body: schema.object({ test: schema.number() }) },
+      options: { body: { accepts: 'application/json' } },
+    },
+    (context, req, res) => res.ok({ body: req.route })
+  );
+  registerRouter(router);
+
+  await server.start();
+  await supertest(innerServer.listener)
+    .post('/')
+    .send({ test: 1 })
+    .expect(200, {
+      method: 'post',
+      path: '/',
+      options: {
+        authRequired: true,
+        tags: [],
+        body: {
+          parse: true, // hapi populates the default
+          maxBytes: 1024, // hapi populates the default
+          accepts: ['application/json'],
+          output: 'data',
+        },
+      },
+    });
+});
+
+describe('body options', () => {
+  test('should reject the request because the Content-Type in the request is not valid', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.post(
+      {
+        path: '/',
+        validate: { body: schema.object({ test: schema.number() }) },
+        options: { body: { accepts: 'multipart/form-data' } }, // supertest sends 'application/json'
+      },
+      (context, req, res) => res.ok({ body: req.route })
+    );
+    registerRouter(router);
+
+    await server.start();
+    await supertest(innerServer.listener)
+      .post('/')
+      .send({ test: 1 })
+      .expect(415, {
+        statusCode: 415,
+        error: 'Unsupported Media Type',
+        message: 'Unsupported Media Type',
+      });
+  });
+
+  test('should reject the request because the payload is too large', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.post(
+      {
+        path: '/',
+        validate: { body: schema.object({ test: schema.number() }) },
+        options: { body: { maxBytes: 1 } },
+      },
+      (context, req, res) => res.ok({ body: req.route })
+    );
+    registerRouter(router);
+
+    await server.start();
+    await supertest(innerServer.listener)
+      .post('/')
+      .send({ test: 1 })
+      .expect(413, {
+        statusCode: 413,
+        error: 'Request Entity Too Large',
+        message: 'Payload content length greater than maximum allowed: 1',
+      });
+  });
+
+  test('should not parse the content in the request', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
+    router.post(
+      {
+        path: '/',
+        validate: { body: schema.buffer() },
+        options: { body: { parse: false } },
+      },
+      (context, req, res) => {
+        try {
+          expect(req.body).toBeInstanceOf(Buffer);
+          expect(req.body.toString()).toBe(JSON.stringify({ test: 1 }));
+          return res.ok({ body: req.route });
```

Comment:
we will have false negatives if change `route` format. shouldn't we pass only data to test?